### PR TITLE
tests: add first approval tests 

### PR DESCRIPTION
**Pull request by matthiasschaub**
- [x] for area calculation choose CRS valid world-wide not just UTM
- [x] Choose threshold for area ratio
- [x] Make approval tests work with parametrized fixtures

Leads to test duplication. Will be fixed in another PR. -> https://github.com/GIScience/sketch-map-tool/issues/456

### CODE REVIEW

File: tests/comparator.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import geopandas
+from approvaltests.core import FileComparator
+
+
+class GeoJSONComparator(FileComparator):
+    def compare(self, received_path: str, approved_path: str) -> bool:
+        if not Path(approved_path).exists() or Path(approved_path).stat().st_size == 0:
+            return False
+
+        # TODO: find optimal UTM for every feature or different projection
```
Comment (koebi) [inline tests/comparator.py:12]: 8857 seems to be an okay projection

**Comment by matthiasschaub** [comment]
For parametrized fixtures in a tests following error gets raised after the first test run (first parameter):
```
    def test_smt_approver(sketch_map_marked_path, vector_path):
        options = (
            Options()
            .with_reporter(SketchMapToolReporter(sketch_map=sketch_map_marked_path))
            .with_comparator(GeoJSONComparator())
        )
        with open(vector_path, "rb") as f:
>           verify_binary(f.read(), ".geojson", options=options)
E           approvaltests.approval_exception.ApprovalException: We noticed that you called verify more than once in the same test. 
E           This is the second call to verify:
E               approved_file: /home/matthias/work/smt/tests/integration/test_approval.test_smt_approver.approved.geojson
E           
E           By default, ApprovalTests only allows one verify() call per test.
E           To find out more, visit: 
E           https://github.com/approvals/ApprovalTests.Python/blob/main/docs/how_to/multiple_approvals_per_test.md
E           
E           # Possible Fixes
E           1. Separate your test into two tests
E           2. In your verify call, add `options=NamerFactory.with_parameters("your_paramater")`
E           3. In your test, call `approvals.settings().allow_multiple_verify_calls_for_this_method()`
```

File: tests/integration/test_approval.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from tests.comparator import GeoJSONComparator
+from tests.reporter import SketchMapToolReporter
+
+fixtures = Path(__file__).parent / "fixtures"
+
+
+@pytest.mark.use_fixtures("vector")
+@pytest.fixture(scope="session")
+def vector_path(tmp_path_factory, uuid_digitize) -> bytes:
+    return tmp_path_factory.getbasetemp() / uuid_digitize / "vector.geojson"
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integration/test_approval.py:15]: Its pytest magic. Its a global fixture defined by pytest. No import needed.

(The other fixtures are also not imported but defined by us in `conftest.py` which is discovered and parsed by pytest magic as well)

[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integration/test_approval.py:15]: test_path_factory gives as a tmp dir unique to the current test run

File: tests/comparator.py (hunk 1 above)
Comment (matthiasschaub) [reply tests/comparator.py:12]: Looks good. I am going to use it.

**Pull request merged**